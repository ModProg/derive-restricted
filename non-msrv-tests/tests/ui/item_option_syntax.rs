#[derive(derive_where::DeriveWhere)]
struct Test1<T>(T);

#[derive(derive_where::DeriveWhere)]
#[derive_where]
struct Test2<T>(T);

#[derive(derive_where::DeriveWhere)]
// Rust itself already fails to parse this and will provide a separate error message.
#[derive_where = invalid]
struct Test3<T>(T);

#[derive(derive_where::DeriveWhere)]
#[derive_where = "invalid"]
struct Test4<T>(T);

#[derive(derive_where::DeriveWhere)]
#[derive_where()]
struct Test5<T>(T);

#[derive(derive_where::DeriveWhere)]
#[derive_where(Debug = "option"; T)]
struct Test6<T>(T);

#[derive(derive_where::DeriveWhere)]
#[derive_where(Debug(); T)]
struct Test7<T>(T);

fn main() {}
