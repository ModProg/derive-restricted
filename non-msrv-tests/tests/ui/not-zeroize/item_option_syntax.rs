#[derive(derive_where::DeriveWhere)]
#[derive_where(skip_inner, Clone)]
struct Test1<T>(core::marker::PhantomData<T>);

#[derive(derive_where::DeriveWhere)]
#[derive_where(Debug = invalid; T)]
struct Test2<T>(T);

fn main() {}
