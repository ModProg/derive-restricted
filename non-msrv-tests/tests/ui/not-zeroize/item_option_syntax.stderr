error: unsupported trait, expected one of expected one of Clone, Copy, Debug, Default, Eq, Hash, Ord, PartialEq, PartialOrd
 --> tests/ui/not-zeroize/item_option_syntax.rs:2:16
  |
2 | #[derive_where(skip_inner, Clone)]
  |                ^^^^^^^^^^

error: unsupported trait syntax, expected one of expected one of Clone, Copy, Debug, Default, Eq, Hash, Ord, PartialEq, PartialOrd
 --> tests/ui/not-zeroize/item_option_syntax.rs:6:24
  |
6 | #[derive_where(Debug = invalid; T)]
  |                        ^^^^^^^
